src/test/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodecUnitTest.java

```diff
@@ -0,0 +1,3336 @@
+ */
+public class GencodeGtfCodecUnitTest extends BaseTest {
+
+    private static final String testResourceDir = publicTestDir + "org/broadinstitute/hellbender/utils/codecs/GENCODE/";
+    private static final String xyzTestFile = largeFileTestDir + "gencode.v26.primary_assembly.annotation.XYZ.gtf";
+    private static final String gencodeHg19TestFile = largeFileTestDir + "gencode.v19.LargeFile.gtf";
+
+    /**
+     * Checks the given feature and the given start and end positions for whether the regions overlap.
+     * @param feature {@link GencodeGtfFeature} to use for overlap checking
+     * @param contig Contig to check for overlap
+     * @param start Interval start position to check for overlap
+     * @param end Interval end position to check for overlap
+     * @return {@code true} if the region in the {@link GencodeGtfFeature} and the given interval overlap, {@code false} otherwise.
+     */
+    private boolean checkForOverlap(final GencodeGtfFeature feature,
+                                    final String contig,
+                                    final int start,
+                                    final int end) {
+
+        boolean overlaps = feature.getChromosomeName().equals(contig);
+
+        // Check for any overlap.
+        // This includes overlapping on either end, as well as either region
+        // being contained within the other.
+        overlaps = overlaps &&
+                ((start >= feature.getStart() ) && (start <= feature.getEnd())) ||
+                ((end >= feature.getStart() ) && (end <= feature.getEnd())) ||
+                ((start <= feature.getStart() ) && (end >= feature.getEnd()));
+
+        return overlaps;
+    }
+
+    /**
+     * Tests that a given query in a file returns the correct number of results
+     * @param contig Contiguous region in the genome in which to search.
+     * @param start Start position within {@code contig} in which to search.
+     * @param end End position within {@code contig} in which to search.
+     * @param numExpectedGenes The number of expected results from the query.
+     * @param testFile A GENCODE GTF {@link File} to query against.
+     */
+    private void testIndexHelper(String contig, int start, int end, int numExpectedGenes, File testFile) {
+        // Now we do our queries:
+        try (FeatureDataSource<GencodeGtfFeature> featureDataSource = new FeatureDataSource<>(testFile) )
+        {
+            final Iterator<GencodeGtfFeature> it = featureDataSource.query( new SimpleInterval(contig, start, end) );
+
+            int geneCount = 0;
+
+            for ( ; it.hasNext() ; ) {
+
+                GencodeGtfFeature feature = it.next();
+
+                // Verify the bounds:
+                Assert.assertTrue( checkForOverlap(feature, contig, start, end) );
+
+                // Keep track of how many genes we've seen:
+                ++geneCount;
+            }
+
+            Assert.assertEquals( geneCount, numExpectedGenes );
```

Comment:
@droazen I thought of this too... I can implement it, but it will add a lot of code to the tests.  I was taking the shortcut of knowing how many results should be returned.  The DataProvider for this has a test for each file that returns all records in each test file, so it would be intractable for that case.  I can shorten the queries and / or files though.  What do you think?